scripts/run-eslint.ts

```diff
@@ -148,99 +208,133 @@ async function runESLint(
  * had increases and decreases. If are were more warnings overall then we fail,
  * otherwise we pass.
  *
- * @param results - The results of running ESLint.
+ * @param args - The arguments.
+ * @param args.results - The results from running `eslint`.
+ * param args.countFileEvenIfNoWarnings - Includes a file in the returned
+ * object even if there are no recorded warnings for it.
+ * param args.fileUpdateStrategy - How to update the warning thresholds file.
+ * @returns True if the number of warnings has increased compared to the
+ * existing number of warnings, false if they have decreased or stayed the same.
  */
-function evaluateWarnings(results: ESLint.LintResult[]) {
+function applyWarningThresholdsQualityGate({
+  results,
+}: {
+  results: ESLint.LintResult[];
+}): QualityGateStatus {
   const warningThresholds = loadWarningThresholds();
-  const warningCounts = getWarningCounts(results);
+  const warningCounts = getWarningCounts({
+    results,
+  });
+
+  const completeWarningCounts = removeFilesWithoutWarnings({
```

Comment:
This is the key difference in this function. If we are checking a selection of files and not the whole project then the set of lint warning counts we've tabulated will be a subset of what is already inside of the warning thresholds file, so we need to merge in what we have rather than completely overwrite the file. We also clean up the file by removing files that actually have no lint warnings whatsoever.